pkg/resolution/resolver/framework/interface.go

```diff
@@ -43,13 +44,27 @@ type Resolver interface {
 	// request and should return an error if any are missing or invalid.
 	ValidateParams(ctx context.Context, params []pipelinev1.Param) error
 
+	// ValidateRequest is given the ressolution request spec
+	// should return an error if the resolver cannot resolve it.
+	// This is currently at an ALPHA stability level and subject to alpha API compatibility policies.
+	ValidateRequest(ctx context.Context, req *v1beta1.ResolutionRequestSpec) error
```

Comment:
Same as the request interface, I realized this will also be breaking (since any previous implementations of this interface will no longer be valid).

Same recommendation here - break this into a new interface or add a breaking change warning.